### PR TITLE
Rename namer kinds

### PR DESCRIPTION
**Pull request by adleong**
Update to match current conventions:

* kind names should not include experimental status
* experimental status of a namer must be explicitly acknowledged by setting `experimental: true` in the namer config
* move namers out of io.l5d package

### CODE REVIEW

File: linkerd/docs/namer.md [hunk 1]
```diff
@@ -105,21 +106,21 @@ linkerd provides support for service discovery via
 [Consul](https://www.consul.io/). Note that this support is still considered
 experimental.
 
-The Consul namer is configured with kind `io.l5d.experimental.consul`, and these parameters:
+The Consul namer is configured with kind `io.l5d.consul`, and these parameters:
 
 * *host* --  the Consul host. (default: localhost)
 * *port* --  the Consul port. (default: 8500)
```
Comment (klingerf) [inline linkerd/docs/namer.md]: Should you mention the experimental parameter here, along with the host and port params?  Or maybe update the text on line 107 to say that the experimental param is required for this namer?

Comment (klingerf) [inline linkerd/docs/namer.md]: (same comment goes for the rest of the experimental namers in this file)

File: namer/consul/src/main/scala/io/buoyant/namer/consul/ConsulInitializer.scala [hunk 2]
```diff
@@ -12,23 +12,27 @@ import io.buoyant.namer.{NamerConfig, NamerInitializer}
  *
  * <pre>
  * namers:
- * - kind: io.l5d.experimental.consul
+ * - kind: io.l5d.consul
  *   host: consul.site.biz
  *   port: 8600
  * </pre>
```
Comment (klingerf) [inline namer/consul/src/main/scala/io/buoyant/namer/consul/ConsulInitializer.scala]: I think this comment should include the experimental config param.

File: namer/consul/src/test/scala/io/buoyant/namer/consul/ConsulTest.scala [hunk 3]
```diff
@@ -20,13 +20,13 @@ class ConsulTest extends FunSuite {
...
                     |host: consul.site.biz
                     |port: 8600
       """.stripMargin
 
     val mapper = Parser.objectMapper(yaml, Iterable(Seq(ConsulInitializer)))
-    val consul = mapper.readValue[NamerConfig](yaml).asInstanceOf[consul]
+    val consul = mapper.readValue[NamerConfig](yaml).asInstanceOf[ConsulConfig]
     assert(consul.host == Some("consul.site.biz"))
     assert(consul.port == Some(Port(8600)))
   }
```
Comment (klingerf) [inline namer/consul/src/test/scala/io/buoyant/namer/consul/ConsulTest.scala]: Maybe add the experimental flag to the yaml, and `assert(!consul.disabled)` -- or alternatively leave the yaml as is and `assert(consul.disabled)`?

**Comment by klingerf** [comment]
⭐  Great, thanks for updating.

File: namer/core/src/main/scala/io/buoyant/namer/NamerInitializer.scala [hunk 4]
```diff
@@ -18,14 +18,23 @@ import io.buoyant.config.ConfigInitializer
...
 @JsonAutoDetect(fieldVisibility = Visibility.PUBLIC_ONLY)
 trait NamerConfig {
   @JsonProperty("prefix")
   var _prefix: Option[Path] = None
 
+  @JsonProperty("experimental")
+  var _experimentalEnabled: Option[Boolean] = None
+
+  @JsonIgnore
+  def experimental = false
```
Comment (olix0r) [inline namer/core/src/main/scala/io/buoyant/namer/NamerInitializer.scala]: this might be clearer as `experimentalRequired`?

Comment (olix0r) [inline namer/core/src/main/scala/io/buoyant/namer/NamerInitializer.scala]: Also, it would be helpful to have doc strings for all 3 of these fields

**Comment by olix0r** [comment]
⭐

**Pull request merged**